network/validator/authorized_sender_validator.go

```diff
@@ -39,8 +40,8 @@ func NewAuthorizedSenderValidator(log zerolog.Logger, slashingViolationsConsumer
 
 // PubSubMessageValidator wraps Validate and returns PubSubMessageValidator callback that returns pubsub.ValidationReject if validation fails and pubsub.ValidationAccept if validation passes.
 func (av *AuthorizedSenderValidator) PubSubMessageValidator(channel channels.Channel) PubSubMessageValidator {
-	return func(from peer.ID, msg interface{}) p2p.ValidationResult {
-		_, err := av.Validate(from, msg, channel, message.ProtocolPublish)
+	return func(from peer.ID, msg *message.Message) p2p.ValidationResult {
+		_, err := av.Validate(from, msg.Payload[0], channel, message.ProtocolPublish)
```

Comment:
same comment about not indexing into `Payload`

Reply:
https://github.com/onflow/flow-go/pull/3831/commits/4616da0d6e5b39e55b5b1b51efa8c50a7ff36f39